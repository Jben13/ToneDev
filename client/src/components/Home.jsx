import React from 'react';
import axios from 'axios';
import {
  Link, Redirect,
} from 'react-router-dom';
import PropTypes from 'prop-types';


class Home extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      loggedIn: true,
    };
    this.testingCookies = this.testingCookies.bind(this);
  }

  testingCookies() {
    const { loggedIn } = this.state;
    axios.get('/api/user/so')
      .then((res) => {
        if (res.data === 'Please Login again') {
          window.alert(res.data);
        }
        this.setState({
          loggedIn: !loggedIn,
        });
      })
      .catch((err) => {
        console.log('line 22', err);
      });
  }

  render() {
    const { loggedIn } = this.state;
    const { username } = this.props;
    if (!loggedIn) {
      return <Redirect to="/login" />;
    }
    return (
      <div>
        <h1>
          Welcome to ToneDev
          {' '}
          {`${username}!!!`}
        </h1>
        {/* <Link to="/orgs">View Top 5 Organizations</Link> */}
        <Link to={{
          pathname: '/orgs',
          state: {
            notify: true,
          },
        }}>
          adsf
        </Link>
        <br />
        <Link to="/repos">View By Repo</Link>
        <br />
        <Link to="/user">User Stats</Link>
        <button type="button" onClick={this.testingCookies}>testingCookies</button>
      </div>
    );
  }
}

Home.propTypes = {
  username: PropTypes.string.isRequired,
};

export default Home;
