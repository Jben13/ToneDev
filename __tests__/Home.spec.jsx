import React from 'react';
import Home from '../client/src/components/Home';

test('App renders component correctly', () => {
  const component = global.shallow(<Home />);
  expect(component).toMatchSnapshot();
});
