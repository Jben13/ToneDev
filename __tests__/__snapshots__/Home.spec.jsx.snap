// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component correctly 1`] = `
<div>
  <h1>
    Welcome to ToneDev
  </h1>
  <Link
    replace={false}
    to="/orgs"
  >
    View Top 5 Organizations
  </Link>
  <br />
  <Link
    replace={false}
    to="/repos"
  >
    View By Repo
  </Link>
  <br />
  <Link
    replace={false}
    to="/user"
  >
    User Stats
  </Link>
</div>
`;
